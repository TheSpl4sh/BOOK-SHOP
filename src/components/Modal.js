import React from 'react'
import "../style/modal-component.scss"

function Modal({ onClose, header, text, actions }) {
    const handleWindowClick = (e) => {
        if (e.target.classList.contains("modal-window")) {
          onClose();
        }
      };
    
    return (
        <>
            <div className="modal-window" onClick={handleWindowClick}>
                <section className="modal-content">
                    <article className='modal-header'>
                        <h2>{header}</h2>
                        <button className='modal-header__close-button' onClick={onClose}>X</button>
                    </article>

                    <article className='modal-text'>
                        <p>{text}</p>
                    </article>

                    <article className="modal-buttons">
                        {actions}
                    </article>
                </section>
            </div>
        </>
    );
}

export default Modal