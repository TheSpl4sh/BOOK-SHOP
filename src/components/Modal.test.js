import "@testing-library/jest-dom/extend-expect"
import {render, screen, fireEvent} from '@testing-library/react';
import Modal from "./Modal";

describe('Modal component', () => {
    it('should render modal component with buttons', () => {
        const handleClose = jest.fn()
        const modalActions = (<button className='modal-buttons__item' onClick={handleClose}>Cancel</button>)
        render(
            <Modal 
                onClose={handleClose} 
                header="Hello world" 
                text="How its going ?"
                actions={modalActions}
            />
        )

        const modalHeader = screen.getByText('Hello world')
        expect(modalHeader).toBeInTheDocument()

        const modalText = screen.getByText('How its going ?')
        expect(modalText).toBeInTheDocument()

        const modalWindow = screen.findAllByText(/modal-window/)

        const modalButton = screen.getByText('Cancel')
        expect(modalButton).toBeInTheDocument()

        fireEvent.click(modalButton)
        expect(handleClose).toHaveBeenCalledTimes(1)

        const modalCloseButton = screen.getByText('X')
        expect(modalCloseButton).toBeInTheDocument()
        
        fireEvent.click(modalCloseButton)
        expect(handleClose).toHaveBeenCalledTimes(2)
    })

    it("should do snapshot", () => {
        const handleClose = jest.fn()
        const modalActions = (<button className='modal-buttons__item' onClick={handleClose}>Cancel</button>)
        
        const modal = render(
            <Modal 
                onClose={handleClose} 
                header="Hello world" 
                text="How its going ?"
                actions={modalActions}
            />)
        expect(modal).toMatchSnapshot();
      });
})