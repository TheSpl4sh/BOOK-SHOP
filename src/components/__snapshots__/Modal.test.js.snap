// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should do snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal-window"
      >
        <section
          class="modal-content"
        >
          <article
            class="modal-header"
          >
            <h2>
              Hello world
            </h2>
            <button
              class="modal-header__close-button"
            >
              X
            </button>
          </article>
          <article
            class="modal-text"
          >
            <p>
              How its going ?
            </p>
          </article>
          <article
            class="modal-buttons"
          >
            <button
              class="modal-buttons__item"
            >
              Cancel
            </button>
          </article>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal-window"
    >
      <section
        class="modal-content"
      >
        <article
          class="modal-header"
        >
          <h2>
            Hello world
          </h2>
          <button
            class="modal-header__close-button"
          >
            X
          </button>
        </article>
        <article
          class="modal-text"
        >
          <p>
            How its going ?
          </p>
        </article>
        <article
          class="modal-buttons"
        >
          <button
            class="modal-buttons__item"
          >
            Cancel
          </button>
        </article>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
